lib/engine/step/g_1817/track.rb

```diff
@@ -15,7 +15,7 @@ def setup
           @hex = nil
         end
 
-        def lay_tile(action, extra_cost: 0, entity: nil)
+        def lay_tile(action, extra_cost: 0, entity: nil, tile_ability: nil)
```

Comment:
does tile_ability always come from action_entity? if it does, you can infer it here instead of passing it in

Reply:
What I'm really passing in is the "when" argument from ability. I could change to only be that. Note: this will be removed completely if I pass in the block instead.

Reply:
Removed.